srml/staking/src/lib.rs

```diff
@@ -514,50 +734,20 @@ impl<T: Trait> Module<T> {
 			});
 		}
 
-		let event = if new_slash_count > grace {
-			let slash = {
-				let base_slash = Self::current_offline_slash();
-				let instances = slash_count - grace;
-
-				let mut total_slash = BalanceOf::<T>::default();
-				for i in instances..(instances + count as u32) {
-					if let Some(total) = base_slash.checked_shl(i)
-							.and_then(|slash| total_slash.checked_add(&slash)) {
-						total_slash = total;
-					} else {
-						// reset slash count only up to the current
-						// instance. the total slash overflows the unit for
-						// balance in the system therefore we can slash all
-						// the slashable balance for the account
-						<SlashCount<T>>::insert(v.clone(), slash_count + i);
-						total_slash = Self::slashable_balance(&v);
-						break;
-					}
-				}
-
-				total_slash
-			};
+		let prefs = Self::validators(&v);
+		let unstake_threshold = prefs.unstake_threshold.min(MAX_UNSTAKE_THRESHOLD);
+		let max_slashes = grace + unstake_threshold;
 
+		let event = if new_slash_count > max_slashes {
+			// They're bailing.
+			let slash = Self::current_offline_slash()
+				// Multiply current_offline_slash by 2^unstake_threshold
+				.checked_shl(prefs.unstake_threshold)
```

Comment:
This should be `unstake_threshold` rather than `prefs.unstake_threshold` right?

`prefs.unstake_threshold` has no upper bound, whereas `unstake_threshold` is guaranteed to be at most `max_unstake_threshold`

Reply:
I have changed it to make my tests work, but we should make sure that I did the right thing here

Reply:
that's right - this should be `unstake_threshold`